src/main/java/educationalproject/programmingstuff/servicies/OrderServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+import educationalproject.programmingstuff.repositories.CommodityItemRepository;
+import educationalproject.programmingstuff.repositories.OrderRepository;
+import educationalproject.programmingstuff.servicies.dto.OrderCreateRequestDto;
+import educationalproject.programmingstuff.servicies.dto.OrderResponseDto;
+import educationalproject.programmingstuff.servicies.dto.RequestedItemDto;
+import educationalproject.programmingstuff.servicies.mappers.OrderResponseMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Validated
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final CommodityItemRepository commodityItemRepository;
+
+    private final OrderRepository orderRepository;
+    private OrderResponseMapper orderResponseMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto book(OrderCreateRequestDto newOrder) {
+        List<RequestedItemDto> requested = newOrder.getRequestedItemDtos();
+        List<CommodityItem> available = getAvailableCommodityItems(requested);
+        return writeOff(requested, available);
+    }
+
+    private List<CommodityItem> getAvailableCommodityItems(List<RequestedItemDto> requestedItemDtos) {
+        Set<Long> itemsId = requestedItemDtos.stream()
+                .map(RequestedItemDto::getItemId)
+                .collect(Collectors.toSet());
+        return commodityItemRepository.getCommodityItemsByItemIdIn(itemsId);
+    }
+
+    private OrderResponseDto writeOff(List<RequestedItemDto> requested, List<CommodityItem> available) {
+        Map<Long, Integer> requestedItemsIdMappedToQuantity = makeMapOf(requested);
+        WriteOffOrThrowException(available, requestedItemsIdMappedToQuantity);
+
+        List<OrderItem> orderItems = available.stream()
+                .map(availableCommodityItem -> {
+                    Item item = availableCommodityItem.getItem();
+                    return OrderItem.builder()
+                            .item(item)
+                            .quantity(requestedItemsIdMappedToQuantity.get(item.getId()))
+                            .build();
+                })
+                .collect(Collectors.toList());
+
+        Order order = Order.builder().orderItems(orderItems).build();
+
+        orderRepository.save(order);
+
+        return orderResponseMapper.makeOrderResponseOf(order);
+    }
+
+    private Map<Long, Integer> makeMapOf(List<RequestedItemDto> commodityItems) {
+        return commodityItems.stream()
+                .collect(Collectors.toMap(
+                        RequestedItemDto::getItemId,
+                        RequestedItemDto::getQuantity));
+    }
+
+    private void WriteOffOrThrowException(
```

Comment:
hmmm, seems that a C# developer helped you write this method:)

Reply:
Done I'm not familiar with any of them. That's totally my invention.  :) Anyway I got rid of it